### PR TITLE
JWT Implementation for Frontend. AppContext for global user and settings states

### PR DESCRIPTION
**Pull request by Jl5her**
# Description

* Uses new authentication method adopted from #62 
* Removes `Type` keyword from types.
* Login form is now a popup and will show when trying to access admin pages without sufficient permission
* AppContext allows accessing user, settings, and socket on any page without fetch requests or initializing. `useContext` 
* Begin standardizing code

Several issues may be resolved by this PR and will be checked after merged.
Fixes #  

## Type of change

- [x] 🐞 Bug fix
* Frontend fixes for new authorization flow presented by the backend changes

- [x] 💡 New feature
* AppContext to access user and settings globally 
* Login Popup
  
# How Has This Been Tested?

- [x] Tested individually.

# Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [x] My changes do not break any features.

### CODE REVIEW

File: backend/src/abacus/contest.ts [hunk 1]
```diff
@@ -12,7 +12,7 @@ class ContestService {
   }
 
   init_aws() {
-    AWS.config.region = 'us-east-2';
+    AWS.config.region = process.env.AWS_REGION || 'us-east-1';
```
Comment (brad-cooley) [inline backend/src/abacus/contest.ts:15]: Why did we change to us-east-1 here?

File: backend/src/api/standings/getStandings.ts [hunk 2]
```diff
@@ -47,7 +47,7 @@ export const getStandings = async (_req: Request, res: Response) => {
           team.problems[problem.id].num_submissions = subs[team.uid][problem.pid].length
           team.problems[problem.id].submissions = subs[team.uid][problem.pid]
           subs[team.uid][problem.pid].every((sub: any) => {
-            if (sub.score > 0) {
+            if (sub.status === "accepted") {
```
Comment (brad-cooley) [inline backend/src/api/standings/getStandings.ts:50]: Why did we change this?

File: frontend/src/pages/admin/users/CreateUser.tsx [hunk 3]
```diff
@@ -1,23 +1,25 @@
-import React, { useState } from "react"
+import React, { ChangeEvent, useState } from "react"
 import { Modal, Form, Input, Select, Button, Message } from "semantic-ui-react"
-import config from '../../environment'
+import config from '../../../environment'
```
Comment (brad-cooley) [inline frontend/src/pages/admin/users/CreateUser.tsx:3]: Not important for this PR, but can we eventually make these relative paths? Or is that not supported?

File: frontend/src/pages/blue/Problems.tsx [hunk 4]
```diff
@@ -1,107 +1,66 @@
...
+          </Table.Header>
+          <Table.Body>
+            {problems ? problems.map((problem: Problem) =>
+              <Table.Row key={problem.pid}>
+                <Table.HeaderCell collapsing textAlign='center'>{problem.id}</Table.HeaderCell>
+                <Table.Cell>
+                  <Link to={`/blue/problems/${problem.id}`}>{problem.name}</Link>
+                </Table.Cell>
+                {/* <Table.Cell></Table.Cell>
+                <Table.Cell></Table.Cell> */}
```
Comment (brad-cooley) [inline frontend/src/pages/blue/Problems.tsx:58]: Do we need this?

**Review by brad-cooley — CHANGES_REQUESTED** [review]

File: backend/src/abacus/contest.ts (hunk 1 above)
Comment (Jl5her) [reply backend/src/abacus/contest.ts:15]: For local testing. Production will be defined in the `AWS_REGION` environment variable and all dummy data will be in us-east-1

File: frontend/src/pages/admin/users/CreateUser.tsx (hunk 3 above)
Comment (Jl5her) [reply frontend/src/pages/admin/users/CreateUser.tsx:3]: Do you mean absolute paths from the root of the project directory?

File: backend/src/api/standings/getStandings.ts (hunk 2 above)
Comment (Jl5her) [reply backend/src/api/standings/getStandings.ts:50]: With the competition date in the future, I was getting negative scores during testing and it was showing as failed on standings.

File: frontend/src/pages/blue/Problems.tsx (hunk 4 above)
Comment (Jl5her) [reply frontend/src/pages/blue/Problems.tsx:58]: Placeholder for the `# of Submissions` and `Latest Submission` columns I omitted from this PR

File: frontend/src/pages/admin/users/CreateUser.tsx (hunk 3 above)
Comment (brad-cooley) [reply frontend/src/pages/admin/users/CreateUser.tsx:3]: Yeah.

**Review by brad-cooley — APPROVED** [review]

**Pull request merged**